packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -400,6 +399,36 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
     `, [userId]);
   }
 
+  async getDialogueRecommendedUsers(userId: string, upvotedUsers:UpvotedUser[], limit = 100): Promise<DbUser[]> {
+  const upvotedUserIds = upvotedUsers.map(user => user._id);
+
+  return this.any(`
+    SELECT u.*
+    FROM unnest($2::text[]) AS uv(_id)
+    INNER JOIN "Users" AS u ON uv._id = u._id
+    WHERE
+      -- Don't recommend you users who've never commented twice your posts OR replied 3 times to your comments 
+      EXISTS (
+        SELECT 2
```

Comment:
This doesn't do what you think it does - it ignores the table you're selecting from and selects a single unnamed row containing the literal value "2". This always exists, so the `EXISTS` clause always returns true. You need to use `COUNT` like below.